### PR TITLE
LPD-20681, LPD-20682 - Creation and item actions flaky tests

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Tasks
[LPD-20681](https://liferay.atlassian.net/browse/LPD-20681) / [LPD-20682](https://liferay.atlassian.net/browse/LPD-20682)

## Goal
- Refactor tests and use robust locators.
- Extract shared Data Set fragment configuration.
- Remove Poshi test covered with current test cases

### CODE REVIEW

**Review by markocikos — APPROVED** [review]
@juanjofgliferay LGTM! :sparkles:

**Comment by juanjofgliferay** [comment]
🤔 Creation actions failure doesn't seem to be related.
 
Screenshot 2024-04-03 at 22 15 17

Item actions may need some more data to represent the Data Set.

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:playwright

File: modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/**
```
Comment (juanjofgliferay) [inline modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts:1]: @markocikos @dsanz I've realized that in a brand new environment (CI) nobody navigated to the Data Sets page, so DB tables and endpoints do not exist. The first test (alphabetical order) that tries to call an API endpoint without navigating through the UI will fail.

To solve this problem, and until we get the DB updated when the portal starts, I've created this `0_` based spec so it navigates to the Data Sets page and setup the environment properly. 

It is a bit hacky, so please, share your thoughts and possible strategies.

Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts:1]: Hmm I'm a bit surprised we didn't run into this issue sooner. I guess the test files are always executed alphabetically, so `creationActions` always did the heavy lifting.

This seems to me like a good case for a new fixture, that we include in all DSM tests. Something like `dataSetManagerTest.ts`. In it, we go to Data Set page and confirm something in UI.

Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts:1]: We usually navigate from the home page to the Data Set using the Control Panel. Loading the Data Sets page does the job.

Now that we are using the API to populate the DB we are skipping some navigation. I'll give a try to the fixture.

Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts:1]: Yeah, but it was curious to me that for example `visualizationModes.spec.ts` never failed for that reason, although it's using only API. That means it always executed after other test files. It's probably alphabetical, i.e. `creationActions.spec.ts` running first and doing the landing on Data Sets page. In theory, we could rely on the alphabetical order, and that is what you did with `0_` prefix, but that might bite us in the butt later down the line. Playwright might change the logic, like implement parallelization on test file level, or we could simply rename files. It's safer imo to run it on every test file.

Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts:1]: The order is alphabetical, see https://playwright.dev/docs/test-parallel#control-test-order. You could try only run `visualizationModes.spec.ts` in a clean DB environment.

Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts:1]: Yeah, it would fail for sure.

**Comment by juanjofgliferay** [comment]
ci:test:sf

File: modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts (hunk 1 above)
Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts]: Got it working with a fixture!! Right now, it is scoped to each test and needs to run after the Login one. Maybe, we could merge the Login with this one.

**Comment by juanjofgliferay** [comment]
ci:test:playwright

**Comment by juanjofgliferay** [comment]
ci:test:relevant

File: modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts (hunk 1 above)
Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/0_dataSetsSetup.spec.ts]: LGTM! I'd only add it to all DSM test files, not just `creationActions`. The reasoning is from above: we might implement parallelization on test file level, or rename files down the line.

> Maybe, we could merge the Login with this one.

No need I think, login is applicable outside of DSM. Composability is one of the main selling point for fixtures after all :slightly_smiling_face:

**Comment by juanjofgliferay** [comment]
Great. Playwright test failures are not related with this PR

**Comment by juanjofgliferay** [comment]
ci:report:2558547845

**Comment by juanjofgliferay** [comment]
Non related failures based on https://github.com/liferay-frontend/liferay-portal/pull/4106#issuecomment-2037803052

**Comment by juanjofgliferay** [comment]
ci:forward:force

**Pull request closed without merging**